File: js/replace-string.js
const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout
});

// Função para perguntar ao usuário e receber a entrada como uma promessa
function ask(question) {
    return new Promise(resolve => readline.question(question, resolve));
}

async function main() {
    const nome = await ask("nome: ");
    const posicaoStr = await ask("posição: ");
    const posicao = parseInt(posicaoStr, 10);

    let nomeModificado = nome.split(''); // Converte a string em um array para modificar

    if (nomeModificado[posicao] === 'R') {
        nomeModificado[posicao] = 'S';
    } else if (nomeModificado[posicao] === 'r') {
        nomeModificado[posicao] = 's';
    } else if (nomeModificado[posicao] === 'M') {
        nomeModificado[posicao] = 'N';
    } else if (nomeModificado[posicao] === 'm') {
        nomeModificado[posicao] = 'n';
    } else {
        nomeModificado.splice(posicao, 1); // Remove o caractere na posição
    }

    nomeModificado = nomeModificado.join(''); // Converte o array de volta para string
    console.log("reformulado:", nomeModificado);

    readline.close();
}

main();

File: cpp/replace-string.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
    string nome;
    int posicao;

    cout << "nome: ";
    cin >> nome;
    cout << "posição: ";
    cin >> posicao;

    if (nome[posicao] == 'R') {
        nome[posicao] = 'S';
    } else if (nome[posicao] == 'r') {
        nome[posicao] = 's';
    } else if (nome[posicao] == 'M') {
        nome[posicao] = 'N';
    } else if (nome[posicao] == 'm') {
        nome[posicao] = 'n';
    } else {
        nome.erase(posicao, 1);
    }

    cout << "reformulado: " << nome << endl;
    return 0;
}
